Fix an issue where MinIO was logging every error twice

nitisht opened the pull request:
## Description
The logging subsystem was initialised under init() method in
both gateway-main.go and server-main.go which are part of
same package. This created two logging targets and hence
errors were logged twice. This PR removes the init() method
from gateway-main.go

## Motivation and Context
Fix error logging on console.

## How to test this PR?
Start MinIO server and create a sitation where MinIO would 
log the error on console. Before this PR, you'd notice two error 
logs per error. With this PR, you should get a single error log
on console

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

Praveenrajmani reviewed (approved):
LGTM

harshavardhana reviewed (changes requested):
I think this was added on purpose can you check @nitisht with `git blame` to ensure that you are not introducing regression?

Commit:
Fix an issue where MinIO was logging every error twice

The logging subsystem was initialized under init() method in
both gateway-main.go and server-main.go which are part of
same package. This created two logging targets and hence
errors were logged twice. This PR removes the init() method
from gateway-main.go

harshavardhana reviewed (approved):
LGTM & tested
